personalwebsite/static/base/dist/js/base.js

```diff
@@ -0,0 +1,85 @@
+var currentPathName = window.location.pathname;
+var currentScreenSize = -1;
+
+/* Handle the different windows sizes here
+ * and only call it once per screen size. */
+function onWindowResize() {
+    var width = window.innerWidth;
+
+    if (width >= 1200 && currentScreenSize != 1200) {
+        onLargeScreen();
+        currentScreenSize = 1200;
+    }
+    else if (width < 1200 && width >= 992 && currentScreenSize != 992) {
+        onMediumScreen();
+        currentScreenSize = 992;
+    }
+
+    else if (width < 992 && width >= 768 && currentScreenSize != 768) {
+        onSmallScreen();
+        currentScreenSize = 768;
+    }
+    else if (width < 768 && width >= 0 && currentScreenSize != 576) {
+        onExtraSmallScreen();
+        currentScreenSize = 576;
+    }
+}
+
+/* Handles events when the screen width is greater or equal to 1200px */
+function onLargeScreen() {
+    showNavBarContent();
+}
+
+/* Handles events when the screen width is greater or equal to 992px
+ * and less than 1200px */
+function onMediumScreen() {
+    showNavBarContent();
+}
+
+/* Handles events when the screen width is greater or equal to 768px
+ * and less than 992px */
+function onSmallScreen() {
+    hideNavBarContent();
+}
+
+/* Handles events when the screen width is greater or equal to 0px
+ * and less than 768px */
+function onExtraSmallScreen() {
+    hideNavBarContent();
+}
```

Comment:
showNavBarContent() and hideNavBarContent() aren't supposed to return the width size, that is being handled on the eventlistener. Currently showNavBarContent() and hideNavBarContent() main purpose is to simply toggle on or off content in the navbar.

I can split this out so that is easier to read what's going on.

Renaming onLargeScreen() to isLargeScreen() wouldn't make sense since they are 2 completely different things. Currently onLargeScreen is getting called when the screen size is of a large screen, the purpose of it is to handle things when the screen changes to a large one.